parsers/filter/filter.parser.ts

```diff
@@ -24,7 +24,10 @@ export default async function (
     const result: InputDataType = [];
     tokens.forEach(token => {
       const valueToCheck = _.get(token, options.key);
-      if (valueToCheck.match(reg)) {
+      if (valueToCheck === undefined) {
+        throw new Error(`key ${options.key} does not exist in token ${token.name}`);
+      }
+      if (String(valueToCheck).match(reg)) {
```

Comment:
A user should be able to filter on a group of tokens even if they are not the same type. So I suggest something like:
```suggestion
      if (!!valueToCheck && String(valueToCheck).match(reg)) {
```

Reply:
That's what I had at first, but I didn't want to change the original behaviour 😅 Anyway I applied the suggested changes.